### PR TITLE
INSTALL: Common Install Script and Miscellaneous Fixes

### PR DESCRIPTION
**Pull request by whispercoros**
Install script changes:
* Added git-lfs
* Attempted to separate dependencies based on the project they belong to
    * **Please double check me on these categories!**
* Generally cleaned things up and made improvements

Readme:
* Updated the getting started and contact information

Sensor control:
* Battery voltage defaults to none when no feedback is received

### CODE REVIEW

**Comment by whispercoros** [comment]
On a side note, these changes are pull requested to master because they were all well tested and cannot really be tested at lake day.

File: install.sh [hunk 1]
```diff
@@ -180,107 +224,190 @@ else
...
+
 #==========================#
 # Finalization an Clean Up #
 #==========================#
 
 # Attempt to build the Navigator stack on client machines
 if !(env | grep SEMAPHORE | grep --quiet -oe '[^=]*$'); then
-    instlog "Building Navigator's software stack with catkin_make"
-    catkin_make -C "$CATKIN_DIR" -j8
+	instlog "Building Navigator's software stack with catkin_make"
```
Comment (DSsoto) [inline install.sh]: It's not necessarily Navigator right? It could be Sub or MIL as a whole.

**Comment by whispercoros** [comment]
Thought I got them all, but nice catch. I'll commit a fix for this and also add a pre-flight check to make sure the script is not run as root. Still need more eyes on the way I split up the dependencies before this is merged.

File: install.sh [hunk 2]
```diff
@@ -81,26 +84,57 @@ while [ "$#" -gt 0 ]; do
...
 			;;
 	esac
 done
 
+if !($INSTALL_ALL || $INSTALL_SUB || $INSTALL_NAV); then
+	echo "A MIL project must be selected for install"
+	echo "Run ./install.sh -h for more information"
+	exit 1
+elif ($INSTALL_ALL); then
+	INSTALL_SUB=true
```
Comment (mattlangford) [inline install.sh]: Should these just be set to true by default so that no extra arguments will install everything?

File: install.sh [hunk 3]
```diff
@@ -81,26 +84,57 @@ while [ "$#" -gt 0 ]; do
...
 # Pre-Flight Check #
 #==================#
 
 instlog "Starting the pre-flight system check to ensure installation was done properly"
 
+# The lsb-release package is critical to check the OS version
+# It may not be on bare-bones systems, so it is installed here if necessary
+sudo apt-get update -qq
+sudo apt-get install -qq lsb-release
+
```
Comment (mattlangford) [inline install.sh]: Can you add all the packages required for SSH, we always run into issues with computers not having the openSSH client/server they need.

File: install.sh [hunk 4]
```diff
@@ -110,25 +144,41 @@ else
...
-#==================================#
+#===================================================#
+# Repository and Set Up and Main Stack Installation #
+#===================================================#
 
 # Make sure script dependencies are installed on bare bones installations
 instlog "Installing install script dependencies"
-sudo apt-get update -qq
-sudo apt-get install -qq lsb-release wget aptitude fakeroot ssh git
+sudo apt-get install -qq wget curl aptitude fakeroot ssh git
```
Comment (mattlangford) [inline install.sh]: `sudo apt-get install openssh-server openssh-client` I think is the command

File: install.sh [hunk 5]
```diff
@@ -180,107 +241,190 @@ else
...
-sudo apt-get install -qq ros-indigo-gazebo7-msgs ros-indigo-gazebo7-ros ros-indigo-gazebo7-plugins ros-indigo-gazebo7-ros-control
-sudo apt-get install -qq ros-indigo-control-toolbox ros-indigo-controller-manager ros-indigo-transmission-interface ros-indigo-joint-limits-interface ros-indigo-hardware-interface
-sudo apt-get install -qq ros-indigo-sophus
+# Visualization and graphical interfaces
+sudo apt-get install -qq libvtk5-dev python-vtk
+sudo apt-get install -qq python-qt4-dev python-qt4-gl
+sudo apt-get install -qq python-opengl freeglut3-dev libassimp-dev
+
+# Tools
+sudo apt-get install -qq git-lfs gitk
```
Comment (mattlangford) [inline install.sh]: Do you not have to run that git lfs setup script that they have on the [website](https://github.com/github/git-lfs/blob/master/INSTALLING.md)? 

`curl -s https://packagecloud.io/install/repositories/github/git-lfs/script.deb.sh | sudo bash`

File: install.sh (hunk 2 above)
Comment (whispercoros) [inline install.sh]: Wasn't sure if people wanted this as a default, but it would be easy to set that up

File: install.sh (hunk 3 above)
Comment (whispercoros) [inline install.sh]: Installing the 'ssh' package should install all necessary client and server related ssh utilities as it is a metapackage for them

File: install.sh (hunk 5 above)
Comment (whispercoros) [inline install.sh]: Double check line 194, I run this command there

**Comment by mattlangford** [comment]
`git lfs install --skip-smudge`

**Comment by mattlangford** [comment]
Can you fix the conflicting file? Then it's okay to merge.
I am okay putting this straight into master since it's mainly updating just the install script (plus a small addition to the battery monitor).

**Comment by mattlangford** [comment]
You'll need to merge in master then recommit - then it should be good

**Comment by ironmig** [comment]
I think Anthony renamed README to readme... Is he using windows or something? 
I did the merge, switching the PR to pull from my branch.

**Comment by whispercoros** [comment]
Wait on this, still changing a few things

**Comment by whispercoros** [comment]
Have you changed the script substantially since I last pulled in changes?

**Comment by ironmig** [comment]
All I did was fix the merge conflict with master

**Pull request merged**